src/lib/datepicker/datepicker-input.ts

```diff
@@ -122,9 +126,14 @@ export class MatDatepickerInput<D> extends CdkDatepickerInput<D> implements Afte
   constructor(
       _elementRef: ElementRef,
       @Optional() _dateAdapter: DateAdapter<D>,
-      @Optional() @Inject(CDK_DATE_FORMATS) _dateFormats: CdkDateFormats,
+      @Optional() @Inject(CDK_DATE_FORMATS) _cdkDateFormats: CdkDateFormats,
```

Comment:
move this param to the end and make it optional (`_cdkDateFormats?`) also put a comment `/** @deletion-target 8.0.0 Make required */`